python/practices/practice-3/src/path2/random_word_selector.py

```diff
@@ -0,0 +1,94 @@
+"""
+This program randomly selects a word from a given ebook. It cleans and
+processes the text, creating a histogram of word frequencies. Then, it selects
+a random word with a probability proportional to its frequency.
+"""
+
+
+import string
+import collections
+import os
+from random import randint
+from bisect import bisect
+from constants.constants import EBOOK
+from helpers.file_reader import read_file
+
+
+def clean_file(file_content):
+    """
+    Read a file and return a histogram of cleaned and processed words.
+
+    Args:
+        file_content (list): A list of text lines from the file.
+
+    Returns:
+        dict: A histogram where keys are cleaned words and values are
+        word frequencies.
+    """
+
+    hist = collections.Counter()
+
+    for line in file_content:
+        line = line.replace('-', ' ')
+
+        for word in line.split():
+            word = word.strip(string.punctuation + string.whitespace)
+            word = word.lower()
```

Comment:
Can we be more compact?
Please refer to: Writing Idiomatic Python 3 - Chapter 2: Working with Data - String

Reply:
Done